vcf-dumper/vcf-dumper-lib/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -74,64 +68,71 @@ public VariantContext transform(Variant variant, Region region) throws CellbaseS
+            prependContextNucleotideToVariant = false;
         }
+
+        // get context nucleotide and add it to the variant
+        String contextNucleotide = getContextNucleotideFromSourceLine(srcLineFields, relativePositionOfContextNucleotide);
+        addContextNucleotideToVariant(variant, contextNucleotide, prependContextNucleotideToVariant);
+
+        return prependContextNucleotideToVariant;
     }
 
-    private String getContextNucleotideFromCellbaseCachingRegions(String chromosome, int contextNucleotidePosition, Region region) throws CellbaseSequenceDownloadError {
-        int regionStart = region.getStart() - 1;
-        if (regionSequence == null) {
-            // if an indel start is the first nucleotide of the region, we will need the previous nucleotide, so we are adding
-            // the preceding nucleotide to the region (region.getStart()-1)
-            int regionEnd = region.getEnd();
-            try {
-                regionSequence = cellbaseClient.getSequence(new Region(chromosome, regionStart, regionEnd));
-            } catch (Exception e) {
-                throw new CellbaseSequenceDownloadError("Error getting from Cellbase sequence for Region " + chromosome + ":"
-                        + regionStart + "-" + regionEnd, e);
-            }
+    private String getContextNucleotideFromSourceLine(String[] srcLineFields, int relativePositionOfContextNucleotide) {
+        String referenceInSrcLine = srcLineFields[3];
+        return referenceInSrcLine.substring(relativePositionOfContextNucleotide, relativePositionOfContextNucleotide + 1);
+    }
+
+    private void addContextNucleotideToVariant(Variant variant, String contextNucleotide, boolean prependContextNucleotideToVariant) {
+        // prepend or append the context nucleotide to the reference and alternate alleles
+        if (prependContextNucleotideToVariant) {
+            variant.setReference(contextNucleotide + variant.getReference());
+            variant.setAlternate(contextNucleotide + variant.getAlternate());
+        } else {
+            variant.setReference(variant.getReference() + contextNucleotide);
+            variant.setAlternate(variant.getAlternate() + contextNucleotide);
         }
-        String nucleotide = getNucleotideFromRegionSequence(contextNucleotidePosition, regionStart, regionSequence);
-        return nucleotide;
     }
 
-    private String getNucleotideFromRegionSequence(int nucleotidePosition, int regionStart, String regionSequence) {
-        int relativePosition = nucleotidePosition - regionStart;
-        return regionSequence.substring(relativePosition, relativePosition + 1);
+    private Variant updateVariantStartAndEnd(Variant variant, boolean contextNucleotideAddedBeforeVariant) {
+        if (contextNucleotideAddedBeforeVariant) {
+            variant.setStart(variant.getStart() - 1);
+        }
+        variant.setEnd(variant.getStart() + variant.getReference().length() - 1);
```

Comment:
The logic for this is not so trivial, depending on whether the variant is an insertion or deletion. Please check https://github.com/opencb/biodata/blob/hotfix/0.4/biodata-models/src/main/java/org/opencb/biodata/models/variant/VariantVcfFactory.java#L72-L80

The new start value does only depend on how the context was added, could the end change be done in a way that doesn't depend on the start coordinate? This issue doesn't seem to affect the export negatively because VCF doesn't report the end coordinate, but the more correct we can get the internal status, the better.